experiments/profile_data.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+
+class ProfileData:
+    """ 
+    Class to hold and organize measurements for run configs
+    """
+
+    def __init__(self):
+        self._data = {}
+
+        self._best_config = None
+        self._best_measurement = None
+
+    def add_measurement(self, run_config, measurement):
+        """
+        Add the measurement for the given run_config
+        """
+        keys = self._extract_run_config_keys(run_config)
+        self._add_measurement_from_keys(keys, measurement)
+
+    def get_measurement(self, run_config):
+        """ 
+        Get the measurement belonging to a given run config
+        """
+        keys = self._extract_run_config_keys(run_config)
+        return self._get_measurement_from_keys(keys)
+
+    def get_config_count(self):
+        """ 
+        Get the total number of model configs in the data
+        """
+        count = 0
```

Comment:
Isn't this just `return len(self._data.keys())`?

Reply:
yes, I used to have a nested n-dimention dictionary that I later cleaned up to always just be 2 layers deep. I was lazy and kept the code because I knew it would still work ;)